rasa/shared/core/training_data/visualization.py

```diff
@@ -403,19 +407,33 @@ def _add_message_edge(
 ) -> None:
     """Create an edge based on the user message."""
 
-    if message:
-        message_key = message.get("intent", {}).get("name", None)
+    if message and message.get("intent", {}).get("name", None):
```

Comment:
I'd suggest extracting `message.get("intent", {}).get("name", None)` to a different variable outside the if/elif block since you're using it twice in this if branch (here and in L421)

This way you can also keep message_key = <the_new_variable> in L426 instead if assigning it to `message.label`.